db/extensions.hh

```diff
@@ -100,6 +100,26 @@ public:
     void add_schema_extension(sstring w, schema_ext_create_func f) {
         _schema_extensions.emplace(std::move(w), std::move(f));
     }
+    /**
+     * A shorthand for the add_schema_extension. Adds a function that adds
+     * the extension to a schema with appropriate constructor overload.
+     */
+    template<typename Extension>
+    void add_schema_extension(sstring w) {
+        add_schema_extension(std::move(w), [] (db::extensions::schema_ext_config cfg) {
+            return std::visit([] (auto v) {
+                return ::make_shared<Extension>(v);
+            }, cfg);
+        });
+    }
+    /**
+     * Similar to the one-argument version, but uses Extension::NAME
+     * as the name of the extension.
+     */
+    template<typename Extension>
+    void add_schema_extension() {
+        add_schema_extension<Extension>(Extension::NAME);
```

Comment:
This seems superfluous, and also assumes magic members, which is confusing when reading the call site. 
Also, proper stl-esque way to do such a thing would be to have a namespace level
```
template<typename T>
struct extension_name_of {
     sstring operator()() const {
        return T::name;
     }
};

// and specialize
template<>
struct extension_name_for<PigletExtension> {
   sstring operator()() const { return ::to_string(time()); }
};
```

But again, superfluous.

Reply:
I see. I removed it.